src/components/tl/ucp/allgather/allgather_knomial.c

```diff
@@ -314,9 +314,20 @@ ucc_status_t ucc_tl_ucp_allgather_knomial_init(ucc_base_coll_args_t *coll_args,
                                                ucc_coll_task_t     **task_h)
 {
     ucc_tl_ucp_team_t *tl_team = ucc_derived_of(team, ucc_tl_ucp_team_t);
-    ucc_rank_t         size    = UCC_TL_TEAM_SIZE(tl_team);
-    ucc_kn_radix_t     radix;
+    ucc_mrange_uint_t *p       = &tl_team->cfg.allgather_kn_radix;
+    ucc_rank_t         tsize   = UCC_TL_TEAM_SIZE(tl_team);
+    ucc_memory_type_t  mtype   = GET_MT(&coll_args->args);
+    size_t             count   = GET_TOTAL_COUNT(&coll_args->args, tsize);
+    ucc_datatype_t     dtype   = GET_DT(&coll_args->args);
+    ucc_kn_radix_t     radix, cfg_radix, opt_radix;
+
+    opt_radix = (mtype == UCC_MEMORY_TYPE_HOST) ? tl_team->opt_radix_host :
```

Comment:
```opt_radix```  ```cfg_radix``` ```radix``` calculation happens multiple times. Can you define a helper function for calculating these variables? you can have a switch/case in this helper function so that you calculate the final ```radix``` based on whether```ucc_min``` or ```ucc_knomial_pattern_get_min_radix```.

Reply:
done